tests/php/View/AttributesHTMLTest.php

```diff
@@ -0,0 +1,290 @@
+            'String' => ['test', 'test', 'String is stored properly as an attribute'],
+            'Int' => ['test', -1, 'Int is stored properly as an attribute'],
+            'Array' => ['test', ['foo' => 'bar'], 'Array is stored properly as an attribute'],
+        ];
+    }
+
+    /** @dataProvider singleAttributeProvider */
+    public function testGetAttribute($name, $value, $message): void
+    {
+        $dummy = new AttributesHTMLTest\DummyAttributesHTML();
+        $this->assertNull(
+            $dummy->getAttribute('non-existent attribute'),
+            'Trying to access a non-existent attribute should return null'
+        );
+
+        $dummy->setAttribute($name, $value);
+        $this->assertSame(
+            $value,
+            $dummy->getAttribute($name),
+            $message
+        );
+
+    }
+
+    public function testGetAttributes(): void
+    {
+        $dummy = new AttributesHTMLTest\DummyAttributesHTML();
+        $dummy->setDefaultAttributes([]);
+        $this->assertSame(
+            [],
+            $dummy->getAttributes(),
+            'When no attributes are set and the default attributes are empty, an empty array should be returned'
+        );
+
+        $dummy->setAttribute('empty', '');
+        $dummy->setAttribute('foo', 'bar');
+        $dummy->setAttribute('Number', 123);
+        $dummy->setAttribute('Array', ['foo' => 'bar']);
+
+        $this->assertSame(
+            [
+                'empty' => '',
+                'foo' => 'bar',
+                'Number' => 123,
+                'Array' => ['foo' => 'bar'],
+            ],
+            $dummy->getAttributes(),
+            'All explicitly defined attributes should be returned'
+        );
+
+        $dummy = new AttributesHTMLTest\DummyAttributesHTML();
+        $dummy->setDefaultAttributes([
+            'foo' => 'Will be overridden',
+            'bar' => 'Not overridden',
+        ]);
+        $this->assertSame(
+            [
+                'foo' => 'Will be overridden',
+                'bar' => 'Not overridden',
+            ],
+            $dummy->getAttributes(),
+            'When no attributes are set and the default attributes are used'
+        );
+
+        $dummy->setAttribute('empty', '');
+        $dummy->setAttribute('foo', 'bar');
+        $dummy->setAttribute('Number', 123);
+        $dummy->setAttribute('Array', ['foo' => 'bar']);
+
+        $this->assertSame(
+            [
+                'foo' => 'bar',
+                'bar' => 'Not overridden',
+                'empty' => '',
+                'Number' => 123,
+                'Array' => ['foo' => 'bar'],
+            ],
+            $dummy->getAttributes(),
+            'Explicitly defined attributes overrides default ones'
+        );
+    }
+
+    public function testAttributesHTML(): void
+    {
+        $dummy = new AttributesHTMLTest\DummyAttributesHTML();
+
+        $dummy->setAttribute('emptystring', '');
```

Comment:
Use a data provider for this instead

Reply:
`getAttributesHTML` takes all the attributes and combined them into one string.

I'm not trying to test individuals scenarios here. I'm trying to test the scenario where you combined all these attributes together.